internal/mode/static/nginx/config/servers.go

```diff
@@ -138,7 +170,14 @@ func createLocations(pathRules []dataplane.PathRule, listenerPort int32) []http.
 		locs = append(locs, createDefaultRootLocation())
 	}
 
-	return locs
+	return locs, matchPairs
+}
+
+// removeSpecialCharacters removes '/', '.' from key and replaces '= ' with 'EXACT',
+// to avoid compilation issues with NJS and NGINX Conf.
+func sanitizeKey(input string) string {
+	s := regexp.MustCompile("[./]").ReplaceAllString(input, "")
+	return regexp.MustCompile("= ").ReplaceAllString(s, "EXACT")
```

Comment:
Is this going to panic if the string doesn't have these characters in it, since you're using `MustCompile`?

Reply:
i'll add unit tests to it